File: aoj/volume100/java/A10011.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;

class A10011 {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int n = scan.nextInt();
        List<Integer> nums = new ArrayList<Integer>();
        for (int i = 0; i < n; i++)
            nums.add(scan.nextInt());

        Collections.reverse(nums);

        System.out.println(join(nums, " "));
    }

    private static <T> String join(List<T> list, String separator) {
        StringBuilder sb = new StringBuilder();

        boolean isFirstItem = true;
        for (T item: list) {
            if (isFirstItem) {
                isFirstItem = false;
            } else {
                sb.append(separator);
            }
            sb.append(item);
        }

        return sb.toString();
    }
}

File: aoj/volume100/nodejs/A10011.js
function main(lines) {
  var nums = lines[1].split(/\s+/).reverse();
  console.log(nums.join(' '));
}

var chunks = [];
process.stdin.on('data', function(chunk) {
  chunks.push(chunk)
});
process.stdin.on('end', function() {
  var lines = chunks.join('').split('\n');
  main(lines);
});
